pkg/controller/migration/convert/network.go

```diff
@@ -93,15 +93,55 @@ func handleCalicoCNI(c *components, install *operatorv1.Installation) error {
 		}
 	}
 
-	// IP
-	if err := c.node.assertEnv(ctx, c.client, containerCalicoNode, "IP", "autodetect"); err != nil {
+	ip, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "IP")
+	if err != nil {
 		return err
 	}
 
-	// IP_AUTODETECTION_METHOD
-	if err := handleAutoDetectionMethod(c, install); err != nil {
+	// IP can be 'autodetect', 'none', or not defined.
+	if ip == nil || *ip == "autodetect" {
+		if err := handleIPAutoDetectionMethod(c, install); err != nil {
+			return err
+		}
+	} else if ip != nil && *ip == "none" {
+		c.node.ignoreEnv(containerCalicoNode, "IP_AUTODETECTION_METHOD")
+	} else {
+		return ErrIncompatibleCluster{
+			err:       fmt.Sprintf("IP=%s is not supported", *ip),
+			component: ComponentCalicoNode,
+			fix:       fmt.Sprintf("remove the IP env var or set it to 'none' or 'autodetect', depending on your cluster configuration"),
+		}
+	}
+
+	ip6, err := c.node.getEnv(ctx, c.client, containerCalicoNode, "IP6")
+	if err != nil {
 		return err
 	}
+	// IP6 can be 'autodetect', 'none', or not defined.
+	if ip6 != nil {
+		if *ip6 == "none" {
+			if err := c.node.assertEnvIsSet(ctx, c.client, containerCalicoNode, "FELIX_IPV6SUPPORT", "false"); err != nil {
+				return err
+			}
+		} else if *ip6 == "autodetect" {
+			if err := handleIPv6AutoDetectionMethod(c, install); err != nil {
+				return err
+			}
+			if err := c.node.assertEnvIsSet(ctx, c.client, containerCalicoNode, "FELIX_IPV6SUPPORT", "true"); err != nil {
```

Comment:
Same here - what if this is set via FelixConfiguration instead of an env var? 

We can probably drop this assertion and not lose much - WDYT?

Reply:
Yeah you're right, this should be changed.

Another option is to use `assertEnv` instead to check that IF `FELIX_IPV6SUPPORT` is defined then it has to be `true`, but maybe we have to assume the starting cluster was valid.

Reply:
We chatted about this in Slack and agreed the best option might be to validate the env var if it is set.